src/OrchardCore/OrchardCore.Users.Core/Authentication/CacheTicketStore.cs

```diff
@@ -0,0 +1,155 @@
+                                .CreateProtector($"{nameof(CacheTicketStore)}_{IdentityConstants.ApplicationScheme}");
+
+        var ticket = DeserializeTicket(dataProtector.Unprotect(data.Value));
+        return ticket;
+    }
+
+    public async Task<string> StoreAsync(AuthenticationTicket ticket)
+    {
+        var key = Guid.NewGuid().ToString();
+
+        var userId = string.Empty;
+        var userName = string.Empty;
+
+        var nameIdentifierClaim = ticket.Principal.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+        var nameClaim = ticket.Principal.FindFirst(ClaimTypes.Name)?.Value ?? ticket.Principal.FindFirst("name")?.Value;
+
+        if (ticket.AuthenticationScheme == IdentityConstants.ApplicationScheme)
+        {
+            userId = nameIdentifierClaim;
+            userName = nameClaim;
+        }
+
+        var dataProtector = _accessor.HttpContext.RequestServices.GetService<IDataProtectionProvider>()
+                                .CreateProtector($"{nameof(CacheTicketStore)}_{IdentityConstants.ApplicationScheme}");
+
+        var ticketValue = dataProtector.Protect(SerializeTicket(ticket));
+        var expiresUtc = ticket.Properties.ExpiresUtc;
+
+        var data = new TicketRecord
+        {
+            Key = key,
+            UserId = userId,
+            Username = userName,
+            LastActivityUtc = DateTime.UtcNow,
+            Value = ticketValue,
+            ExpiresUtc = expiresUtc,
+        };
+
+        var cache = _accessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        var bytes = Serialize(data);
+        await cache.SetAsync($"{_keyPrefix}-{key}", bytes, new DistributedCacheEntryOptions() { AbsoluteExpiration = expiresUtc.Value });
+
+        return key;
+    }
+
+    private byte[] SerializeTicket(AuthenticationTicket source)
+        => TicketSerializer.Default.Serialize(source);
+
+    private AuthenticationTicket DeserializeTicket(byte[] source)
+        => source == null ? null : TicketSerializer.Default.Deserialize(source);
+
+    private byte[] Serialize(TicketRecord m)
+    {
+        using (var ms = new MemoryStream())
+        {
+            (new DataContractSerializer(typeof(TicketRecord))).WriteObject(ms, m);
+            return ms.ToArray();
+        }
+    }
+
+    private TicketRecord Deserialize(byte[] byteArray)
+    {
+        using (var ms = new MemoryStream(byteArray))
+        {
+            return (TicketRecord)(new DataContractSerializer(typeof(TicketRecord))).ReadObject(ms);
+        }
+    }
+}
+
+[Serializable]
+[DataContract]
+public record class TicketRecord
+{
+    [DataMember]
+    public string Key { get; set; }
+    [DataMember]
+    public string UserId { get; set; }
+    [DataMember]
+    public string Username { get; set; }
+    [DataMember]
+    public DateTime LastActivityUtc { get; set; }
+    [DataMember]
+    public byte[] Value { get; set; }
+    [DataMember]
+    public DateTimeOffset? ExpiresUtc { get; set; }
```

Comment:
Jsut DateTime since it's UTC

Reply:
data type was originally aligned with Authentication Properties's ExpireUtc. - Updated to DateTime now